### PR TITLE
Add support for weaviate multi tenancy

### PR DESCRIPTION
**Pull request by mbennett-talentnet**
This PR adds the option to specify a tenant in the weaviate client configuration if the weaviate class uses the new multi-tenancy option.

### CODE REVIEW

File: langchain/src/vectorstores/weaviate.ts [hunk 1]
```diff
@@ -126,6 +130,7 @@ export class WeaviateStore extends VectorStore {
 
       const flattenedMetadata = flattenObjectForWeaviate(document.metadata);
       return {
+        ...this.tenant ? { tenant: this.tenant } : {},
```
Comment (jacoblee93) [inline langchain/src/vectorstores/weaviate.ts:133]: `...this.tenant` looks like a typo?

**Comment by jacoblee93** [comment]
Thanks for this! One question

File: langchain/src/vectorstores/weaviate.ts (hunk 1 above)
Comment (jacoblee93) [reply langchain/src/vectorstores/weaviate.ts:133]: I think you mean 
```typescript
...(this.tenant ? { tenant: this.tenant } : {}),
```

Comment (mbennett-talentnet) [reply langchain/src/vectorstores/weaviate.ts]: Not a typo but I do agree it's more readable with the parenthesis. I've updated as per your suggestion.

**Pull request merged**